File: ValidAnagram.cpp
/****************************************
 *
 *  242. Valid Anagram solution c++
 *
 *  Joel M. Gonzalez Rodriguez
 *
*/***************************************

bool isAnagram(string s, string t)
{
    int l, r;
    l = r = 0;

    if (s.length() != t.length())
        return false;

    sort(s.begin(), s.end());
    sort(t.begin(), t.end());

    while (l < s.length() && r < t.length())
    {
        if (s[l] != t[r])
            return false;

        l++;
        r++;
    }

    return true;
}

File: Python/Easy/Valid_Anagram.py
'''

    242. Valid Anagram

    Solution:

'''

def populate_dict(s: string, hm: dict):
    for i in s:
        if hm.get(i) == None:
            hm[i] = 1

        else:
            hm[i] += 1

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        dict_1, dict_2 = {}

        populate_dict(s, dict_1)
        populate_dict(t, dict_2)

        return dict_1.items() == dict_2.items()